### ISSUE TITLE
IsReference return 1 when called on all forms with Compiler Override

### ISSUE THREAD

**Issue by llde** [llde/xOBSE#2]

**Comment by llde** [comment]
I cannot reproduce this issue. CLosing

**Comment by Forlini91** [comment]
The problem is caused by the OBSE compiler:

```
scn fnTestRef

String_Var name
Ref rVar

Begin Function { name, rVar }
	If IsReference rVar
		Print name + " is a Reference"
	Else
		Print name + " is a Base Object"
	EndIf
End
```

This works correctly, but if I enable the OBSE compiler with '_' before Function, the command break: it always returns 1, no matter the parameter (as opposed to my first report, where I only reported the forms which can have references).

**Comment by llde** [comment]
It seems more an issue  with the compiler override  then with the function then. maybe it.pass parameters  differently. Reopened.

**Comment by llde** [comment]
@Forlini91 do the issue happen also with let statement?

**Comment by Forlini91** [comment]
Yes, it happens with Let too. I tested with this:
```
scn fnTestRef

String_Var name
Ref rVar
Short isRef

Begin _Function { name, rVar }
	Let isRef := IsReference rVar
	If isRef
		Print name + " is a Reference"
	Else
		Print name + " is a Base Object"
	EndIf
End
```
Everything (base objects without references, base objects with references and references) return 1, so it prints "... is a Reference"

**Comment by llde** [comment]
Ok Tested in a _GameMode block (so Compiler override active):
First result in the first script execution is returning correctly 0, from second it is returning always 1.
In the next iterations return always 1.
Maybe there is an issue with the ExtractArgs() hook and the script's ContextExecution retriving.

**Comment by llde** [comment]
@Forlini91  @shadeMe  
I fixed the specific functions, but it may be a greater issue with compiler override that affect similar functions. In case open a new bug.
Closing this.

**Comment by llde** [comment]
Fixed in 2f06997
